src/frontend/src/lib/components/signer/SignerPermissions.svelte

```diff
@@ -0,0 +1,136 @@
+	let confirm: PermissionsConfirmation | undefined;
+	$: confirm = $payload?.confirm;
+
+	/**
+	 * During the initial UX review, it was decided that permissions should not be permanently denied when "Rejected," but instead should be ignored.
+	 * This means that if the user selects "Reject," the permission will be requested again the next time a similar action is performed.
+	 * This approach is particularly useful since, for the time being, there is no way for the user to manage their permissions in the Oisy UI.
+	 */
+	const ignorePermissions = () => {
+		if (isNullish(confirm)) {
+			toastsError({
+				msg: { text: $i18n.signer.permissions.error.no_confirm_callback }
+			});
+			return;
+		}
+
+		confirm([]);
+
+		resetPrompt();
+	};
+
+	const approvePermissions = () => {
+		if (isNullish(confirm)) {
+			toastsError({
+				msg: { text: $i18n.signer.permissions.error.no_confirm_callback }
+			});
+			return;
+		}
+
+		confirm((scopes ?? []).map((scope) => ({ ...scope, state: ICRC25_PERMISSION_GRANTED })));
+
+		resetPrompt();
+	};
+
+	const onReject = () => ignorePermissions();
+
+	const onApprove = () => approvePermissions();
+
+	const listItems: Record<
+		IcrcScopedMethod,
+		{
+			icon: ComponentType;
+			label: string;
+		}
+	> = {
+		icrc27_accounts: {
+			icon: IconWallet,
+			label: replaceOisyPlaceholders($i18n.signer.permissions.text.icrc27_accounts)
+		},
+		icrc49_call_canister: {
+			icon: IconShield,
+			label: $i18n.signer.permissions.text.icrc49_call_canister
+		}
+	};
+
+	let requestAccountsPermissions = false;
+	$: requestAccountsPermissions = nonNullish(
+		scopes?.find(({ scope: { method } }) => method === ICRC27_ACCOUNTS)
+	);
+</script>
+
+{#if nonNullish(scopes) && nonNullish($payload)}
+	<form in:fade on:submit|preventDefault={onApprove} method="POST">
+		<h2 class="mb-4 text-center">{$i18n.signer.permissions.text.title}</h2>
+
+		<SignerOrigin payload={$payload} />
+
+		<div class="mb-6 rounded-lg border border-light-blue bg-light-blue p-6">
+			<p class="break-normal font-bold">{$i18n.signer.permissions.text.requested_permissions}</p>
+
+			<ul class="mt-2.5 flex list-none flex-col gap-1">
+				{#each scopes as scope}
+					{@const { icon, label } = listItems[scope.scope.method]}
+
+					<li class="flex items-center gap-2 break-normal pb-1.5">
+						<svelte:component this={icon} size="24" />
+						{label}
+					</li>
+				{/each}
+			</ul>
+		</div>
+
+		{#if requestAccountsPermissions}
+			<div class="mb-10 flex gap-4 rounded-lg border border-dust bg-white p-4">
+				<IconAstronautHelmet />
+
+				<div>
+					<label class="block text-sm font-bold" for="ic-wallet-address"
+						>{$i18n.signer.permissions.text.your_wallet_address}</label
+					>
+
+					<output id="ic-wallet-address" class="break-all"
+						>{shortenWithMiddleEllipsis({ text: $icrcAccountIdentifierText ?? '' })}</output
+					>
+				</div>
+			</div>
+		{/if}
+
+		<ButtonGroup>
+			<button type="button" class="error block flex-1" on:click={onReject}
```

Comment:
isn't `block` the default? with or without is the same?

TBD for me: include this CSS class into `Button` component